### PR TITLE
refactor(angular): improve pagination module

### PR DESCRIPTION
**Pull request by jeripeierSBB**

### CODE REVIEW

File: src/angular/migration-guide.md [hunk 1]
```diff
@@ -168,17 +180,62 @@ The link module has been integrated into the button module. All button types can
...
-_Automatic migration available_
+_Partial migration available_
 
-The contextmenu module has been integrated into the new more generic menu module, which also supports submenus.
-Due to this change you have to define the contextmenu trigger yourself (the migration will take care of it),
-and the contextmenu no longer depends on `SbbDropdown`. The APIs of `SbbDropdown` and `SbbMenu` have some differences,
-so please check your code carefully after migration.
-The width of a `sbb-menu-item` is not fixed anymore and adapts itself to the content. The width also can be overridden.
+The `sbb-pagination` component has been removed. From now on, only the `sbb-paginator` component can be used.
+At the sbb-pagination component, the `length` property stood for the number of pages. With the sbb-paginator component,
```
Comment (kyubisation) [inline src/angular/migration-guide.md:188]:
```suggestion
With the sbb-pagination component, the `length` property stood for the number of pages. With the sbb-paginator component,
```

File: src/angular/migration-guide.md [hunk 2]
```diff
@@ -168,17 +180,62 @@ The link module has been integrated into the button module. All button types can
...
+}
+```
+
+The `SbbPageChangeEvent` is now called `SbbPageEvent` and contains different properties than before.
+You have to manually migrate them.
+
+#### Styles
+
+`sbb-paginator` and `sbb-navigation` don't reserve space around them anymore.
+Please control the layout yourself.
```
Comment (kyubisation) [inline src/angular/migration-guide.md:236]:
```suggestion
Please manually check the layout.
```

File: src/angular/pagination/navigation/navigation.scss [hunk 3]
```diff
@@ -4,20 +4,20 @@ $navigationIconTextMargin: 4;
 
 .sbb-navigation {
   display: flex;
+  margin: 0;
```
Comment (kyubisation) [inline src/angular/pagination/navigation/navigation.scss:7]: Is there a need to manually set the margin to 0? Do we inherit it from somewhere?

File: src/angular/pagination/paginator/paginator.scss [hunk 4]
```diff
@@ -1,63 +1,80 @@
...
+
+$paginatorDisabledColor: $sbbColorSilver;
+$paginatorDisabledColorLean: $sbbColorStorm;
+
 .sbb-paginator {
   display: flex;
   padding-left: 0;
   list-style: none;
-  margin-left: 0;
+  margin: 0;
```
Comment (kyubisation) [inline src/angular/pagination/paginator/paginator.scss:15]: Is there a need to manually set the margin to 0? Do we inherit it from somewhere?

File: src/angular/schematics/ng-add/migrations/pagination-migration.ts [hunk 5]
```diff
@@ -32,12 +32,13 @@ export class PaginationMigration extends Migration<null, DevkitContext> {
   private _handlePagination(element: MigrationElement) {
```
Comment (kyubisation) [inline src/angular/schematics/ng-add/migrations/pagination-migration.ts:32]: Maybe change the link on line 26?

File: src/showcase-merge/app/angular/icon-overview/cdn-icon-list/cdn-icon-list.component.html [hunk 6]
```diff
@@ -18,5 +18,9 @@
 <div [class.sbb-icon-fit]="filterForm.get('fitIcons').value">
   <sbb-cdn-icon *ngFor="let cdnIcon of filteredIcons | async" [cdnIcon]="cdnIcon"></sbb-cdn-icon>
 </div>
-<sbb-paginator [pageSize]="pageSize"></sbb-paginator>
+<sbb-paginator
+  [pageSize]="pageSize"
+  aria-label="Select page of icons"
```
Comment (kyubisation) [inline src/showcase-merge/app/angular/icon-overview/cdn-icon-list/cdn-icon-list.component.html:23]: I'm not sure about this. Does the following sound "smoother"?
```suggestion
  aria-label="Select icon page"
```

**Review by kyubisation — CHANGES_REQUESTED** [review]
Thanks for the double check and the fixes!
I added a few comments.

File: src/angular/pagination/paginator/paginator.scss (hunk 4 above)
Comment (jeripeierSBB) [reply src/angular/pagination/paginator/paginator.scss:15]: Yes from ul / li styles. I will add a comment do clarify

File: src/angular/pagination/navigation/navigation.scss (hunk 3 above)
Comment (jeripeierSBB) [reply src/angular/pagination/navigation/navigation.scss:7]: Yes from ul / li styles. I will add a comment do clarify

File: src/angular/schematics/ng-add/migrations/pagination-migration.ts (hunk 5 above)
Comment (jeripeierSBB) [reply src/angular/schematics/ng-add/migrations/pagination-migration.ts:32]: Thanks for the reminder :-)

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**